client/modules/workplace/actions/preview.js

```diff
@@ -102,7 +100,33 @@ export default {
       });
       if (neededFields.filter(f => f.id).length) result = fields;
     }
+
     return result;
+
+    function determineFieldsBySavedConstructors(fieldsArr, constructors) {
```

Comment:
please, don't mutate function parameter `fieldsArr` http://eslint.org/docs/rules/no-param-reassign
It is better to create a clone and make manipulations on it